src/platform/arm-nuttx/iotjs_module_gpio-arm-nuttx-stm32f.inl.h

```diff
@@ -0,0 +1,445 @@
+  GPIO_PORTB,
+  GPIO_PORTC,
+  GPIO_PORTD,
+  GPIO_PORTE,
+  GPIO_PORTF,
+  GPIO_PORTG
+};
+
+
+uint32_t gpioPin[GPIO_MAX_PINNO] = {
+  GPIO_PIN0,
+  GPIO_PIN1,
+  GPIO_PIN2,
+  GPIO_PIN3,
+  GPIO_PIN4,
+  GPIO_PIN5,
+  GPIO_PIN6,
+  GPIO_PIN7,
+  GPIO_PIN8,
+  GPIO_PIN9,
+  GPIO_PIN10,
+  GPIO_PIN11,
+  GPIO_PIN12,
+  GPIO_PIN13,
+  GPIO_PIN14,
+  GPIO_PIN15
+};
+
+
+uint32_t gpioMode[6] = {
+  0,  // none
+  GPIO_PULLUP,
+  GPIO_PULLDOWN,
+  GPIO_FLOAT,
+  GPIO_PUSHPULL,
+  GPIO_OPENDRAIN
+};
+
+
+namespace iotjs {
+
+
+// GPIO implementeation for arm-nuttx target.
+class GpioArmNuttxStm32f : public Gpio {
+ public:
+  explicit GpioArmNuttxStm32f(JObject& jgpio);
+
+  static GpioArmNuttxStm32f* GetInstance();
+
+  virtual int Initialize(GpioReqWrap* gpio_req);
+  virtual int Release(GpioReqWrap* gpio_req);
+  virtual int SetPin(GpioReqWrap* gpio_req);
+  virtual int WritePin(GpioReqWrap* gpio_req);
+  virtual int ReadPin(GpioReqWrap* gpio_req);
+  virtual int SetPort(GpioReqWrap* gpio_req);
+  virtual int WritePort(GpioReqWrap* gpio_req);
+  virtual int ReadPort(GpioReqWrap* gpio_req);
+
+  bool _initialized;
+};
+
+
+GpioArmNuttxStm32f::GpioArmNuttxStm32f(JObject& jgpio)
+    : Gpio(jgpio)
+    , _initialized(false) {
+}
+
+
+GpioArmNuttxStm32f* GpioArmNuttxStm32f::GetInstance()
+{
+  return static_cast<GpioArmNuttxStm32f*>(Gpio::GetInstance());
+}
+
+
+#define GPIO_WORKER_INIT_TEMPLATE \
+  GpioArmNuttxStm32f* gpio = GpioArmNuttxStm32f::GetInstance(); \
+  IOTJS_ASSERT(gpio->_initialized == false); \
+  GpioReqWrap* gpio_req = reinterpret_cast<GpioReqWrap*>(work_req->data); \
+  GpioReqData* req_data = gpio_req->req(); \
+
+
+void AfterWork(uv_work_t* work_req, int status) {
+  GpioArmNuttxStm32f* gpio = GpioArmNuttxStm32f::GetInstance();
+
+  GpioReqWrap* gpio_req = reinterpret_cast<GpioReqWrap*>(work_req->data);
+  GpioReqData* req_data = gpio_req->req();
+
+  if (status) {
+    req_data->result = kGpioErrSys;
+  }
+
+  JArgList jargs(2);
+  jargs.Add(JVal::Number(req_data->result));
+
+  switch (req_data->op) {
+    case kGpioOpInitize:
+    {
+      if (req_data->result == kGpioErrOk) {
+        gpio->_initialized = true;
+      }
+      break;
+    }
+    case kGpioOpRelease:
+    {
+      if (req_data->result == kGpioErrOk) {
+        gpio->_initialized = false;
+      }
+      break;
+    }
+    case kGpioOpSetPin:
+    case kGpioOpWritePin:
+    {
+      break;
+    }
+    case kGpioOpReadPin:
+    {
+      if (req_data->result == kGpioErrOk) {
+        jargs.Add(JVal::Bool(!!req_data->value));
+      }
+      break;
+    }
+    case kGpioOpSetPort:
+    case kGpioOpWritePort:
+    {
+      break;
+    }
+    case kGpioOpReadPort:
+    {
+      if (req_data->result == kGpioErrOk) {
+        jargs.Add(JVal::Number(req_data->value));
+      }
+      break;
+    }
+    default:
+    {
+      IOTJS_ASSERT(!"Unreachable");
+      break;
+    }
+  }
+
+  MakeCallback(gpio_req->jcallback(), *Gpio::GetJGpio(), jargs);
+
+  delete work_req;
```

Comment:
This ``delete work_queue`` should be here? Like [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) , in order to manage codes well, I think isn't it better to put this expression to where work_req is acquired